### PR TITLE
Add Bash Logic/Math Operators 

### LINKED ISSUE #297: Add Bash Logic/Math Operators
# Description

Logic/Math Operators for Bash aren't presently on Code Thesaurus. This issue is to add that concept to the system.

# Requirements

* Copy `web/thesauruses/_meta/operators.json` to the right Bash directory in `web/thesauruses/`
* Edit the operators.json file to accurately reflect the latest major version of Bash
* Delete the `"name":` fields before your PR
* Add in `"comment":` if you wish to explain something in the code
* Add in `"not-implemented": "true"` if the concept doesn't exist in the language. You can then leave `"code":` blank.

# Claiming This Issue

Please leave a comment below so I can assign you to it. This is to prevent multiple people from working on the same issue. 

Check out [CONTRIBUTING.md](https://github.com/codethesaurus/codethesaur.us/blob/main/CONTRIBUTING.md) for more information on how to contribute. 

Check out [https://docs.codethesaur.us/thesaurus/](https://docs.codethesaur.us/thesaurus/) for information on how Code Thesaurus works and how to add to the thesauruses.

# Need Help?

Leave a comment on here with your question, or feel free to ask [@geekygirlsarah](https://twitter.com/geekygirlsarah) on Twitter. You can also email coreteam@codethesaur.us too.

# Thanks!

We really value and appreciate the additions people make to Code Thesaurus, no matter how large or small. So thank you for taking the time to work on this!

### PR DESCRIPTION
**Pull request by Blastoise**
## What GitHub issue does this PR apply to?

Resolves #297 

## What changed and why?

Added Bash Operators

## Checklist

- [X] I claimed any associated issue(s) and they are not someone else's
- [X] I have looked at documentation to ensure I made any revisions correctly
- [X] I tested my changes locally to ensure they work
- [ ] (If editing Django) I have added or edited any appropriate unit tests for my changes

## Any additional comments or things to be aware of while reviewing?

NA

### CODE REVIEW

**Comment by Blastoise** [comment]
Read one of your comments on the issues about not just adding the operator but the whole method. Fixing and making a pull request again.

**Comment by Blastoise** [comment]
@geekygirlsarah  I have updated the file after reading your comments on other people issues. Kindly review the PR.

File: web/thesauruses/bash/operators.json [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+      "left_shift",
+      "left_shift_assignment",
+      "right_shift",
+      "right_shift_assignment"
+    ],
+    "Conditional Operators": ["ternary", "null_forgiving"]
+  },
+  "operators": {
+    "addition": {
+      "code": "(( 8+4 ))"
```
Comment (geekygirlsarah) [inline web/thesauruses/bash/operators.json:57]: I'm curious about the parenthesis. are they required to do the operations?

File: web/thesauruses/bash/operators.json [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+      "code": "${a:-20}"
+    },
+    "is": {
+      "not-implemented": "true"
+    },
+    "is_not": {
+      "not-implemented": "true"
+    },
+    "and": {
+      "code": "[[ $a == 'A' && $b == 'B' ]]"
```
Comment (geekygirlsarah) [inline web/thesauruses/bash/operators.json]: There's a lot going on here. Could you simplify it (like `expression && expression`)? I want to make sure it's easily comparable to other languages, and don't want it to be too confusing.

**Comment by geekygirlsarah** [comment]
I think this is close. Want to clarify/fix those? Then it should be good to go!

File: web/thesauruses/bash/operators.json (hunk 1 above)
Comment (Blastoise) [reply web/thesauruses/bash/operators.json:57]: Yes it is required. This is called **Arithmetic evaluation/expansion** in bash. Basically every mathematical operation needs to done inside it.
Refer this: https://program-script.com/bash/syntax/arithmetic.php

File: web/thesauruses/bash/operators.json (hunk 2 above)
Comment (Blastoise) [reply web/thesauruses/bash/operators.json]: Sure. I just wrote the expression explicitly.

**Comment by Blastoise** [comment]
I have made the necessary changes.

File: web/thesauruses/bash/operators.json (hunk 1 above)
Comment (geekygirlsarah) [reply web/thesauruses/bash/operators.json:57]: I haven't seen it before (but also haven't written or seen a ton of Bash). Thanks!

**Comment by geekygirlsarah** [comment]
Thanks for the info and edits! I'll merge it in.

Be sure to sign up for Hacktoberfest if you want this to count towards the shirt/stickers!

File: web/thesauruses/bash/operators.json (hunk 1 above)
Comment (Blastoise) [reply web/thesauruses/bash/operators.json:57]: No. It's the built-in way to do it. Some people use commands for achieving it.

**Comment by Blastoise** [comment]
> Thanks for the info and edits! I'll merge it in.
> 
> Be sure to sign up for Hacktoberfest if you want this to count towards the shirt/stickers!

Thanks. It was great working with you.

**Comment by Blastoise** [comment]
@geekygirlsarah You can merge it in. All checks have passed.

**Comment by geekygirlsarah** [comment]
Huh, I usually do it right after my "thanks" message and not sure why it didn't happen. (Forgot? Distracted?)

**Comment by Blastoise** [comment]
> Huh, I usually do it right after my "thanks" message and not sure why it didn't happen. (Forgot? Distracted?)

Not an issue.

**Comment by geekygirlsarah** [comment]
> 
> 
> > Huh, I usually do it right after my "thanks" message and not sure why it didn't happen. (Forgot? Distracted?)
> 
> Not an issue.

Well not any more, I merged it! 😂

**Pull request merged**